### PR TITLE
fix: ignore not proper files from repo config storage

### LINKED ISSUE #1241: Issue: Artipie checks dotfiles when loading yaml/yml files
Hi Team,

**Artipie version**: Docker image v0.28.0
**Platform**: Kubernetes (K3S) v1.26.1 on Windows 11 

After successfully experimenting with docker-compose I wanted to test running Artipie on a local Kubernetes cluster using the file storage backend. I created a custom Helm chart with the following additional templates:

- a ConfigMap for the artipie.yml configuration file,
- a ConfigMap for the _storages.yaml file,
- a Secret for _credentials file.

These configuration files are then mounted in the Artipie pod as follows:

```
containers:
        - name: {{ .Chart.Name }}
          securityContext:
            {{- toYaml .Values.securityContext | nindent 12 }}
          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
          imagePullPolicy: {{ .Values.image.pullPolicy }}
          ports:
            - name: http
              containerPort: {{ .Values.service.http.targetPort }}
              protocol: TCP
            - name: swagger
              containerPort: {{ .Values.service.api.targetPort }}
              protocol: TCP
          livenessProbe:
            httpGet:
              path: /.health
              port: http
          readinessProbe:
            httpGet:
              path: /
              port: http
          resources:
            {{- toYaml .Values.resources | nindent 12 }}
          volumeMounts:
            - name: main
              mountPath: "{{ .Values.artipie.config_dir }}"
            - name: artipie-data
              mountPath: "{{ .Values.artipie.data_dir }}"
            - name: configs
              mountPath: "{{ .Values.artipie.data_dir }}/configs"
            - name: repos
              mountPath: "{{ .Values.artipie.data_dir }}/configs/{{ .Values.artipie.default_org }}"
      volumes:
        # main configuration file for Docker
        - name: main
          configMap:
            name: artipie
            items:
              - key: "artipie"
                path: "artipie.yml"
        # artifacts storage root directory
        - name: artipie-data
          hostPath:
            path: "/<localFolder>/data"
            type: Directory
        # Additional config files
        - name: configs
          projected:
            sources:
              # Storage aliases (per repo configuration)
              - configMap:
                  name: storages
                  items:
                    - key: "storages"
                      path: "_storages.yaml"
              # default user credentials
              - secret:
                  name: credentials
                  items:
                    - key: "credentials"
                      path: "_credentials.yaml"
                  optional: false
```
This caused an issue when starting Artipie because it analyzed hidden folders/files creating for bind mounting the ConfigMaps and Secrets:

```
 〉kubectl logs -n artipie artipie-84c4d5bd5-vlp6r                                                                                                                 Mon Feb  6 10:57:25 2023
[main] INFO com.artipie.VertxMain - Used version of Artipie: v0.28.0
[main] INFO org.eclipse.jetty.util.log - Logging initialized @3397ms to org.eclipse.jetty.util.log.Slf4jLog
[main] INFO org.reflections.Reflections - Reflections took 996 ms to scan 1 urls, producing 36 keys and 78 values
[main] INFO com.artipie.asto.factory.Storages - Initiated storage factory [type=fs, class=FileStorageFactory]
[main] INFO com.artipie.VertxMain - Artipie was started on port 8080
[Thread-10] INFO com.artipie.asto.fs.FileStorage - Found 24 objects by the prefix "" in /var/artipie/configs by /var/artipie/configs: [..2023_02_06_09_57_10.2284382907/_credentials.yaml, ..2023_02_06_09_57_10.2284382907/_storages.yaml, _credentials.yaml, _storages.yaml, private/..2023_02_06_09_57_10.3087496895/binaries.yaml, private/..2023_02_06_09_57_10.3087496895/docker-remote.yaml, private/..2023_02_06_09_57_10.3087496895/docker.yaml, private/..2023_02_06_09_57_10.3087496895/helm.yaml, private/..2023_02_06_09_57_10.3087496895/maven-remote.yaml, private/..2023_0Exception in thread "main" java.util.concurrent.CompletionException: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalStateException: Filename `..2023_02_06_09        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:649)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture.postFire(CompletableFuture.java:614)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:653)
        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.IllegalStateException: Filename `..2023_02_06_09_57_10` should have `.yaml` or `.yml` extension or be without extension
        at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051)
        at com.google.common.cache.LocalCache.get(LocalCache.java:3951)
        at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3974)
        at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4935)
        at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4941)
        at com.artipie.settings.repo.RepositoriesFromStorage.config(RepositoriesFromStorage.java:99)
        at com.artipie.VertxMain.lambda$startRepos$1(VertxMain.java:206)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at com.artipie.VertxMain.lambda$startRepos$3(VertxMain.java:208)
        at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
        ... 5 more
Caused by: java.lang.IllegalStateException: Filename `..2023_02_06_09_57_10` should have `.yaml` or `.yml` extension or be without extension
        at com.artipie.settings.ConfigFile.valueFrom(ConfigFile.java:123)
        at com.artipie.settings.repo.RepositoriesFromStorage$FilesContent.configContent(RepositoriesFromStorage.java:162)
        at com.artipie.settings.repo.RepositoriesFromStorage$1.load(RepositoriesFromStorage.java:55)
        at com.artipie.settings.repo.RepositoriesFromStorage$1.load(RepositoriesFromStorage.java:52)
```

Would it be possible to ignore dotfiles to prevent thees kind of issues ?

### PR DESCRIPTION
**Pull request by olenagerasimova**
closes #1241 

Fixed `ConfigFile` to check extension of hidden files properly. On start Artipie scans configs directory looking for repository configs and tries to create repositories from `.yml` and `.yaml` files. If yaml file content is invalid, it writes an exception into log and continue the work. 
Note, that in the case, when hidden file is valid repository config, the repository will be started and will have path with dot, for example: `artipie/.my-hidden-bin-repo`.
Basically, we do not expect users to have any other files except for repository and artipie configs in the configuration storage.

### CODE REVIEW

**Review by dgarus — APPROVED** [review]
Great!
lgtm
